server/datastore/mysql/software.go

```diff
@@ -15,13 +15,25 @@ import (
 	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
 	"github.com/fleetdm/fleet/v4/server/fleet"
 	"github.com/go-kit/kit/log/level"
+	"github.com/google/uuid"
 	"github.com/jmoiron/sqlx"
 )
 
+type softwareIDChecksum struct {
+	ID       uint   `db:"id"`
+	Checksum string `db:"checksum"`
+}
+
 // Since DB may have millions of software items, we need to batch the aggregation counts to avoid long SQL query times.
 // This is a variable so it can be adjusted during unit testing.
 var countHostSoftwareBatchSize = uint64(100000)
 
+// Since a host may have a lot of software items, we need to batch the inserts.
+// The maximum number of software items we can insert at one time is governed by max_allowed_packet, which already be set to a high value for MDM installers,
```

Comment:
@roperzh Is the `max_allowed_packet` a residue from the MDM PoC 2 years ago? (I remember adding it for the PoC and maybe we are not relying on such a thing?)

Reply:
unfortunately we still need it because we were forced to store bootstrap packages in the database :(

for installers we won and we're using S3

Reply:
Ah thanks!, I can update the comments on another PR.